packages/next/src/server/base-server.ts

```diff
@@ -702,6 +702,15 @@ export default abstract class Server<ServerOptions extends Options = Options> {
   ): Promise<void> {
     await this.prepare()
     const method = req.method.toUpperCase()
+
+    // Extract the w3c trace context headers and pass them to tracer
+    const traceParentHeader = req.headers.traceparent
```

Comment:
IMHO, it'd be better to either move the propagation extraction code from below to here or to pass headers to the `trace` call. For instance, this code could execute:

```
spanContext = propagation.extract(context.active(), req.headers)
```

And pass an optional `spanContext` to the `trace()` method. This way the SDK can configure whatever propagation strategy it likes outside of Next and it'd work.

Reply:
If we don't want to pass `spanContext` to `trace()`, there should also be a way to propagate it via OpenTelemetry's `context.with()`.

Reply:
Here's the alternative, which, IMHO, should cover the target cases better: https://github.com/vercel/next.js/pull/57084

Reply:
OK thanks, I was planning to work on your feedback today but I'm happy to let you take it from here with your PR

Can you help me understand how your PR is an improvement / expand on what you mean by `This way the SDK can configure whatever propagation strategy it likes outside of Next and it'd work`?

Reply:
In a otel setup (in an application, not in Next), you could do something like this:

```
traceProvider.register({
  propagator: new XPropagator(),
})
```

E.g. you could use [`AWSXRayPropagator`](https://github.com/open-telemetry/opentelemetry-js-contrib/blob/main/propagators/opentelemetry-propagator-aws-xray/src/AWSXRayPropagator.ts) or make your own.

This propagator could work off headers or some special environment context. Next.js wouldn't need to know any of that - it'd just call a generic `propagator.extract()` and  (eventually)`propagator.inject()` and delegate this part fully to the configuration of the application or a deployment environment.

Reply:
And FWIW, your PR is super close, it just limits which propagators can be used.

Reply:
I see, thanks for explaining and for opening your PR. I'll close this one out in favor of yours